ludwig/schema/hyperopt/executor.py

```diff
@@ -0,0 +1,100 @@
+from dataclasses import field
+from typing import Dict, Optional, Union
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.hyperopt.scheduler import BaseSchedulerConfig, SchedulerDataclassField
+
+
+@dataclass
+class ExecutorConfig(schema_utils.BaseMarshmallowConfig):
+    """Basic executor settings."""
+
+    type: str = schema_utils.StringOptions(options=["ray"], default="ray", allow_none=False)
+
+    num_samples: int = schema_utils.PositiveInteger(
+        default=10,
+        description=(
+            "This parameter, along with the space specifications in the parameters section, controls how many "
+            "trials are generated "
+        ),
+    )
+
+    time_budget_s: int = schema_utils.PositiveInteger(
+        default=3600, description="The number of seconds for the entire hyperopt run."
+    )
+
+    cpu_resources_per_trial: int = schema_utils.PositiveInteger(
+        default=1, description="The number of CPU cores allocated to each trial"
+    )
+
+    gpu_resources_per_trial: int = schema_utils.NonNegativeInteger(
+        default=0, description="The number of GPU devices allocated to each trial"
+    )
+
+    kubernetes_namespace: Optional[str] = schema_utils.String(
+        default=None,
+        description=(
+            "When running on Kubernetes, provide the namespace of the Ray cluster to sync results between "
+            "pods. See the Ray docs for more info."
+        ),
+    )
+
+    max_concurrent_trials: Union[str, int, None] = schema_utils.OneOfOptionsField(
+        default="auto",
+        allow_none=True,
+        description=("The maximum number of trials to train concurrently. Defaults to auto if not specified."),
+        field_options=[
+            schema_utils.PositiveInteger(
+                default=1, allow_none=False, description="Manually set a number of concurrent trials."
+            ),
+            schema_utils.StringOptions(
+                options=["auto"],
+                default="auto",
+                allow_none=False,
+                description="Automatically set number of concurrent trials.",
+            ),
+        ],
+    )
+
+    scheduler: BaseSchedulerConfig = SchedulerDataclassField(description="")
+
+
+def ExecutorDataclassField(description: str, default: Dict = {}):
+    allow_none = False
```

Comment:
May be cleaner to put this as an optional argument with a default if we want to allow the option for None, otherwise, I think it may be better to just put None in the return below. WDYT?

Reply:
Same as above